src/TemplateProcessor.ts

```diff
@@ -1119,13 +1129,24 @@ export default class TemplateProcessor {
         return didSet; //true means that the data was new/fresh/changed and that subsequent updates must be propagated
     }
 
+    /**
+     * This
+     * @param output
+     * @param jsonPtr
+     * @param data
+     * @param op
+     * @private
+     */
     private async setUntrackedLocation(output, jsonPtr, data, op:"set" |"setDeferred"| "delete"="set") {
         if(op==="delete"){
             if(!jp.has(this.output, jsonPtr)){
                 return; // we are being asked to remove something that isn't here
             }
+            jp.remove(output, jsonPtr); // delete the node from output
+            if (data === TemplateProcessor.NOOP || data === null) data = undefined;
```

Comment:
is NOOP really allowed to get in here? NOOP is supposed to prevent operations from actually occurring. Hmm.
and what is up w weird spacing/formatting of this code?

Reply:
yeah, I overthought it. Only should care of null, which is exactly triggers TemplateProcessor.NOOP default value in one place. But should not propagate it here